### DISCUSSION TITLE
The `Const()` in ROHD require explicit width declaration

### DISCUSSION THREAD

**Discussion by quekyj** [intel/rohd#335] [Q&A]
The `Const()` Logic in ROHD required explicit width declaration when width greater than one. I am not sure whether this is a bug? Shouldn't that `Const(4)` throw an error since this is invalid due to its actually `Const(4, width: 1)` which shows value of 0 instead. It's hard to identify the bug when the design is big, and the user accidentally forgets to assign the width. 

I am thinking that we can create a feature that helps them resolve the width at the backend. Or throw an error/warning when user perform this kind of behavior. At least better than current silently make it become `0`. Or is there already existing function built that I am not aware?

```dart
import 'package:rohd/rohd.dart';

void main() {
 final val10 = Const(10).value.toInt(); 
 print(val10);
}
```

https://dartpad.dev/?id=70c8dc936001270a03dac68c92a44f06

**Comment by chykon** [comment]
I agree, things like this can be pretty hard to debug if you're not aware of this behavior.

Related issue: https://github.com/intel/rohd/issues/142

**Reply by mkorbel1** [reply]
Thanks @quekyj for bringing it up.  Yes #142 tracks improving this, thanks @chykon.
